docs/pages/extensibility/ui.md

```diff
@@ -39,8 +39,94 @@ Meshery leverages remote providers for identity management. These providers can
 
 The user avatar behavior, which changes based on the user's status, can be customized by extending the [User Component](https://github.com/meshery/meshery/blob/7de49ef4928f114080f923f2ad261f4433ca91d6/ui/components/User.js#L46).
 
+### Extensibility: Permissions (CASL)
 
-###  Build-Time UI Extensibility
+Meshery UI uses [CASL.js](https://casl.js.org/v6/en/) to implement it's permissions framework. CASL is an isomorphic authorization JavaScript library which restricts what resources a given client is allowed to access.
+It has been designed to be very customizable and robust.
+It uses keys as it's unit checks to see if a permission to perform or action or not.
+
+#### Implementation
+
+Once a user has logged in, the backend will send a response with the permissions that the user has. Those permissions will be used to check abilities on the frontend.
+
+```JSON
+Example keys response
+
+{"keys": [
+    {
+      "id": "382da488-9a92-4a5b-958d-c4bfe1e80253",
+      "function": "View All Users",
+    },
+    {
+      "id": "fa7de118-2d08-4b07-b9d7-3e0baead6d04",
+      "function": "View Profile",
+    },
+]}
+```
+
+Meshery UI uses `permissions_constants.js` file to keep record of all the keys to check against (fully customizable if self hosting meshery). The keys are accessible using monikers (properties) of the keys object and have been setup this way to make devEx better. Anytime any key
```

Comment:
Always capitalize Meshery.

Reply:
Don't use "devEx"

Reply:
Don't describe monikers as properties.

Reply:
How does the permission framework and its customizability change if Meshery is self-hosted?